packages/next-swc/crates/next-core/src/router_source.rs

```diff
@@ -145,12 +146,18 @@ impl ContentSource for NextRouterContentSource {
                     ContentSourceContent::Rewrite(rewrite.build()).cell().into(),
                 )
             }
-            RouterResult::FullMiddleware(data) => ContentSourceResultVc::exact(
+            RouterResult::Middleware(data) => ContentSourceResultVc::exact(
                 ContentSourceContent::HttpProxy(
                     ProxyResult {
-                        status: data.headers.status_code,
-                        headers: data.headers.headers.clone(),
-                        body: data.body.clone().into(),
+                        status: data.status_code,
+                        headers: data.headers.clone(),
+                        body: Body::from_stream(data.body.read().map(|chunk| {
+                            chunk.map_err(|e| {
+                                BodyError::new(
+                                    "error streaming proxied contents: ".to_string() + e.as_str(),
```

Comment:
Use `format!` here instead.

Otherwise, `thiserror::Error` allows you to indicate the provenance of your error in a way that can preserve backtraces with `#[source]`, so it would be even better to do this without any concatenation.

Reply:
Updated. I'm struggling to serialize `anyhow::Error`, so for now this is just a string error message.